src/MomentoTaggedCache.php

```diff
@@ -49,9 +48,31 @@ public function get($key, $default = null): mixed
     /**
      * @throws UnknownError
      */
-    public function flush()
+    public function flush(): bool
     {
-        throw new UnknownError("flush operations is currently not supported.");
+        $tags = $this->tags->getNames();
+        if (!self::validateTags($tags)) {
+            throw new UnknownError("Empty tag is not valid. Please provide non-empty tags to flush.");
+        }
+        foreach ($tags as $tag) {
+            $keys = $this->store->setFetch($tag);
+            if (!$keys) {
+                return false;
+            } else {
+                foreach ($keys as $key) {
```

Comment:
This loop body looks correct. As far as performance, here we have 2 commands per key in the tag set. We can cut this down by doing bulk deletes on the keys (eg call delete on 1000 keys at a time) and then deleting the set itself (setDelete).

Reply:
Like this?
```
foreach ($keys as $key) {
    $forgetResult = $this->store->forget($key);
    if (!$forgetResult) {
        return false;
     } 
}
$setDeleteResult = $this->store->setDelete($tag);
if (!$setDeleteResult) {
    return false;
}
```

Reply:
Makes sense! I should have implemented `setDelete` instead of `setRemoveElement` in our PHP SDK.
Now I need to add `setDelete` so that I can implement the above.

Reply:
Yes that's what I had in mind. I also intended to use a bulk delete, where you can issue a delete to many keys with a single call. That command variant is still on the backlog, so bulk delete can wait. Ket's do `setDelete` though.